benchmarks/iperf/src/opnsense/scripts/iperf/ruby_iperf.rb

```diff
@@ -182,15 +186,20 @@ def run_test_thread(interface = 'any')
       until connection.closed?
         begin
           command = connection.gets.strip.split(' ')
-          case command.shift
+          case command[0]
           when 'start'
             interface = 'any'
-            if command.length > 0
-              intf = command.shift
+            port = 0
+            if command.length > 1
+              intf = command[1]
               # check if a valid interface was given
               interface = intf if intf =~ /^[a-z0-9_-]+$/
             end
-            data = run_test_thread interface
+            if command.length > 2
+              port = command[2].to_i
+              port = 0 if !find_open_ports.include? port
```

Comment:
Ruby: `unless condition` is equal to `if !(condition)`. It makes it easier to read.

```suggestion
              port = 0 unless find_open_ports.include? port
```

Reply:
changed now as recommended